app/apps/client/RealAppsEngineUIHost.js

```diff
@@ -0,0 +1,42 @@
+import { Meteor } from 'meteor/meteor';
+import { Session } from 'meteor/session';
+import { AppsEngineUIHost } from '@rocket.chat/apps-engine/client/AppsEngineUIHost';
+
+import { APIClient } from '../../utils';
+import { getUserAvatarURL } from '../../utils/lib/getUserAvatarURL';
+
+export class RealAppsEngineUIHost extends AppsEngineUIHost {
+	constructor() {
+		super();
+
+		this._baseURL = document.baseURI.slice(0, -1);
+	}
+
+	async getClientRoomInfo() {
```

Comment:
async getClientRoomInfo(rid)

Reply:
We cannot apply this change for a couple of reasons:

* This class extends the `AppsEngineUIHost` class from the apps-engine, which defines the `getClientRoomInfo` and `getClientUserInfo` methods _without any parameters on the signature_

* The `externalComponents` use the `AppsEngineUIClient` to communicate with the Rocket.Chat window, and _it does not have access to either room id or user id to provide to this function call_